chapters/en/chapter5/evaluation.mdx

```diff
@@ -0,0 +1,387 @@
+    torch_dtype = torch.float16
+else:
+    device = "cpu"
+    torch_dtype = torch.float32
+
+pipe = pipeline(
+    "automatic-speech-recognition",
+    model="openai/whisper-small",
+    torch_dtype=torch_dtype,
+    device=device,
+)
+```
+
+Next, we'll load the Dhivehi test split of Common Voice 13. You'll remember from the previous section that the Common
+Voice 13 is *gated*, meaning we had to agree to the dataset terms of use before gaining access to the dataset. We can
+now link our Hugging Face account to our notebook, so that we have access to the dataset from the machine we're currently
+using.
+
+Linking the notebook to the Hub is straightforward - it simply requires entering your Hub authentication token when prompted.
+Find your Hub authentication token [here](https://huggingface.co/settings/tokens) and enter it when prompted:
+
+```python
+from huggingface_hub import notebook_login
+
+notebook_login()
+```
+
+Great! Once we've linked the notebook to our Hugging Face account, we can proceed with downloading the Common Voice
+dataset. This will take a few minutes to download and pre-process, fetching the data from the Hugging Face Hub and
+preparing it automatically on your notebook:
+
+```python
+from datasets import load_dataset
+
+common_voice_test = load_dataset(
+    "mozilla-foundation/common_voice_13_0", "dv", split="test"
+)
+```
+
+Evaluating over an entire dataset can be done in much the same way as over a single example - all we have to do is **loop**
+over the input audios, rather than inferring just the first one. We first define a helper function that yields examples
+from our dataset using a generator object, and then pass the generator through the pipeline to get our predictions and
+references. The following code cell will take approximately five minutes if running on a GPU with half-precision, peaking
+at 12GB memory:
+
+```python
+from tqdm import tqdm
+
+
+def data(dataset):
+    """Helper function that wraps a dataset as a generator object for compatibility with pipeline"""
+    for i, item in enumerate(dataset):
+        yield {**item["audio"], "reference": item["sentence"]}
+
+
+all_predictions = []
+all_references = []
+
+# run streamed inference
+for out in tqdm(
+    pipe(
+        data(common_voice_test),
+        max_new_tokens=128,
+        generate_kwargs={"task": "transcribe", "language": "sinhalese"},
+        batch_size=32,
+    ),
+    total=len(common_voice_test),
+):
+    all_predictions.append(out["text"])
+    all_references.append(out["reference"][0])
+```
+
+<Tip>
+    If you experience a CUDA out-of-memory (OOM) when running the above cell, incrementally reduce the `batch_size` by
+    factors of 2 until you find a batch size that fits in your device.
+</Tip>
+
+And finally, we can compute the WER. Let's first compute the orthographic WER, i.e. the WER without any post-processing:
+
+```python
+from evaluate import load
+
+wer_metric = load("wer")
+
+wer_ortho = 100 * wer_metric.compute(
```

Comment:
Why do we switch to percentages? (multiplying by 100)

Reply:
For consistency with our previous scripts (e.g. the Whisper event) where we returned the WER in % (see https://huggingface.co/spaces/whisper-event/leaderboard?dataset=mozilla-foundation%2Fcommon_voice_11_0&config=ar&split=test)

Agnostic as to whether we do it in absolute or percentage, but the latter is more consistent with the scripts we've published in the past